test/actor_tests.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+    Copyright 2023 Adobe
+    Distributed under the Boost Software License, Version 1.0.
+    (See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+
+/**************************************************************************************************/
+
+// identity
+#include <stlab/concurrency/actor.hpp>
+
+// boost
+#include <boost/test/unit_test.hpp>
+
+// stlab
+#include <stlab/concurrency/default_executor.hpp>
+#include <stlab/concurrency/utility.hpp>
+
+/**************************************************************************************************/
+
+void increment_by(int& i, int amount) { i += amount; }
+
+void increment(int& i) { increment_by(i, 1); }
+
+template <class T>
+T get_actor_value(stlab::actor<T>& a) {
+    return stlab::await(a.send([](auto& x) { return x; }));
+}
+
+/**************************************************************************************************/
+
+BOOST_AUTO_TEST_CASE(actor_construct_with_arguments) {
+    stlab::actor<int> a(stlab::default_executor, "actor_int", 42);
+    stlab::future<void> f = a.send([](auto& i) { BOOST_REQUIRE(i == 42); });
+
+    stlab::await(f);
+
+    BOOST_REQUIRE(get_actor_value(a) == 42);
+}
+
+/**************************************************************************************************/
+
+BOOST_AUTO_TEST_CASE(actor_construct_void) {
+    stlab::actor<void> a(stlab::default_executor, "actor_void");
+    std::atomic_bool sent{false};
```

Comment:
Is there a reason to make this an atomic_bool?
I think there is a natural sync with the following await.

Reply:
Yes, you're right. Fixed.